### ISSUE TITLE
OpenShift v2 End of Life'd

### ISSUE THREAD

**Issue by Nexuist** [Nexuist/CoinMarketCapAPI#21]
Hello everyone,

On September 30, 2017, OpenShift shut down their V2 online cloud hosting platform. This is what CoinMarketCapAPI was running on. If you try to access it now, you will get an HTTP 503 error. OpenShift is offering a new version of their platform but that will require some migration on my part to get the existing codebase to be compatible with it.

I am opening this issue to allow anyone who is interested to voice their opinion on the future of this project and provide any feedback they may have for me. If enough people want it, I will take the necessary steps to get this working on OpenShift again, or migrate to another platform like DigitalOcean.

Thank you all for your support,
Andi

**Comment by Cagy** [comment]
It was easy to switch to: https://coinmarketcap.com/api/
Thanks for your time!

**Comment by Imthebest01** [comment]
@Cagy did you get Coinmarketcap official API with abbreviations? You can only request with full crypto name ex: https://coinmarketcap.com/bitcoin

that really bothers me

**Comment by Cagy** [comment]
No, I store the labels per API, and symbols etc in a DB somewhere and fetch the right label when needed.

**Comment by altsheets** [comment]
Please keep on updating your code.

We have been very happy users.  
Eventually actually ran our own instance of your code on our own server.  

But now, that started crashing 2 days ago, and I have not had the time yet to look into why.  
Any idea?  Has coinmarketcap.com changed its layout?

> provide any feedback

Keen to hear my opinion? --> Continue please!

Thank you very much.

**Comment by Nexuist** [comment]
Hi @altsheets. It has now been a year or two of you providing consistent pull requests for this project for which I am grateful for.

I am considering reviving the project and beginning to host it again. However, you seem like the primary user for this software and so you should get a say in what it does.

I would like to ask:

* What features do you want to be added?
* What features do you not use? 
* Would you consider paying a small monthly fee to keep the software maintained and the server online? This way, you do not have to worry about maintaining the software whenever it breaks.

I will try to get a look at your PRs before the end of the month. Thank you again for your effort.

**Comment by altsheets** [comment]
Thx a lot!
Good to hear from you.

About my pull requests: Sure. When I make improvements, I make them public,
so others can benefit too, no problem at all.
I am grateful that you provided the code, that makes everything much easier.

We are running your code on our own server anyways.

Costs: I suggest you have a look at scaleway, then it only costs you 23.88
EUR  per year.

Nice to have features:
* out of the box already the first 3 pages, not only the first 100 coins
but 300.
(But it's there already in a way, like I mentioned in that issue: simply
run the service three times, on three different ports, for cmc pages 1-3)
* tunable polling time perhaps?

no other features in mind. your thing is rather complete already.

keep up the good work!

:-)
